gems/smithy/lib/smithy/vise/enum_shape.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Vise
+    # Represents an enum shape from the model.
+    class EnumShape < Shape
+      # (see Shape#initialize)
+      def initialize(id, shape)
+        super
+        @members = parse_members(shape['members'])
+      end
+
+      # @return [Hash<String, MemberShape>]
+      attr_reader :members
+
+      private
+
+      def parse_members(members)
```

Comment:
A general naming discussion - I'm not sure I would call this "parsing".  I'd actually call this "building" - we're taking generic ruby data (Hashs) and using it to build a structured model.